frontend/src/model.js

```diff
@@ -359,6 +359,43 @@ function ListUsers() {
   })
 }
 
+function CreateUser(user) {
+  var url = 'manager/create_user';
+  request('put', url, user
+  ).then(response => {
+    if(response.status === 200) {
+      console.log("user added:", response.data);
+      let returned_user = response.data;
+      setState(['users', 'index', returned_user.Username], returned_user);
+      clearState(['app', 'users', 'email']);
```

Comment:
Because this bit of state is related to UX (e.g. form input) it is generally kept out of this `model` namespace. The state manipulation in here largely deals with interactions with the back-end and putting (or clearing) the returned state / updates into the portion of the app-db that reflect what is provided by the underlying system.

For bits that refer to front-end-only logic, the management of that state is kept next to the components that use them. Typically the way this is done is by providing a `successCallback` and `errorCallback` as the last two arguments to this function, and then calling those callbacks (if they're non-nill ... e.g. `successCallback && successCallback(response.data)` and then in the component that makes the call here, it passes in an anonymous function (`() => {clearState(['app', 'users', 'email'])}`) to this function to perform the state update after the response is returned. That way the management of UX state doesn't leak into this model class which shouldn't have to care about those bits of data.

Reply:
gotcha